_posts/2025-02-04-data-wants-to-be-free.md

```diff
@@ -0,0 +1,306 @@
+---
+layout: post
+title: "Data wants to be free: fast data exchange with Apache Arrow"
+description: ""
+date: "2025-02-04 00:00:00"
+author: David Li, Ian Cook, Matt Topol
+categories: [application]
+image:
+  path: /img/arrow-result-transfer/part-1-share-image.png
+  height: 1200
+  width: 705
+---
+
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+<style>
+.a-header {
+  color: purple;
+}
+.a-data {
+  color: blue;
+}
+.a-length {
+  color: orange;
+}
+.a-padding {
+  color: red;
+}
+</style>
+
+_This is the second in a series of posts that aims to demystify the use of
+Arrow as a data interchange format for databases and query engines._
+
+As data practitioners, we often find our data “held hostage”. Instead of being
+able to use data as soon as we get it, we have to spend time—time to parse and
+clean up inefficient CSV files, time to wait for an outdated query engine to
+struggle with a few gigabytes of data, and time to wait for the data to make
+it across a socket. It’s that last point we’ll focus on today. In an age of
+multi-gigabit networks, why is it even a problem in the first place? And it is
+a problem—research by Mark Raasveldt and Hannes Mühleisen in their [2017
+paper](https://ir.cwi.nl/pub/26415/p852-muehleisen.pdf) demonstrated that some
+systems take over **ten minutes** to transfer a dataset that should only take
+ten *seconds*.
+
+Why are we waiting 60 times as long as we need to? [As we've argued before,
+serialization overheads plague our
+tools](https://arrow.apache.org/blog/2025/01/10/arrow-result-transfer/)—and
+Arrow can help us here. So let’s make that more concrete: we’ll compare how
+PostgreSQL and Arrow encode the same data to illustrate the impact of the data
+serialization format. Then we’ll tour various ways to build protocols with
+Arrow, like Arrow HTTP and Arrow Flight, and how you might use each of them.
+
+# PostgreSQL vs Arrow: Data Serialization
+
+Let’s compare the [PostgreSQL binary
```

Comment:
Is the PostgreSQL binary format different from the data serialization format used in the PostgreSQL wire protocol? If it is, I'm not sure the comparison between it and Arrow is fair. If it is, I think a couple sentences to draw the user in would be good.

> When you connect to a PostgreSQL database  and execute a query, PostgreSQL sends the result back to you using the PostgreSQL wire protocol. The data in your query result is encoded within this protocol using the PostgreSQL binary format.

(Note: The above could be false but it's the basic thing I was after)

Reply:
Also, for where we mention "PostgreSQL wire protocol" lower down, I think we should add a link to each instance, such as to https://www.postgresql.org/docs/current/protocol.html.

Reply:
In this context, I'm pretty sure the `PostgreSQL binary format` **is** the `PostgreSQL wire protocol`